File: Java/src/main/java/com/interviewkata/Child.java
package com.interviewkata;

public class Child {
    private Gift desiredGift;
    private Gift obtainedGift;
    private String name;

    public String showMeWhatYouGot() {
        return "Soy " + name + " y recibi un " + obtainedGift.getDescription();
    }

    public Gift getDesiredGift() {
        return desiredGift;
    }

    public void setDesiredGift(Gift desiredGift) {
        this.desiredGift = desiredGift;
    }

    public Gift getObtainedGift() {
        return obtainedGift;
    }

    public void setObtainedGift(Gift obtainedGift) {
        this.obtainedGift = obtainedGift;
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
}

File: Kotlin/src/main/java/com/interviewkata/Child.kt
package com.interviewkata

class Child {
    var desiredGift: Gift? = null
    var obtainedGift: Gift? = null
    var name: String? = null

    fun showMeWhatYouGot() = "Soy " + name + " y recibi un " + obtainedGift!!.description
}
